Mage.Sets/src/mage/cards/k/KjeldoranGuard.java

```diff
@@ -0,0 +1,149 @@
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.common.FilterLandPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile, Susucr
+ */
+public final class KjeldoranGuard extends CardImpl {
+
+    private static final FilterLandPermanent filter = new FilterLandPermanent();
+
+    static {
+        filter.add(SuperType.SNOW.getPredicate());
+    }
+
+    public KjeldoranGuard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{W}");
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // {T}: Target creature gets +1/+1 until end of turn. When that creature leaves the battlefield this turn, sacrifice Kjeldoran Guard. Activate only during combat and only if defending player controls no snow lands.
+        CompoundCondition condition = new CompoundCondition(
+                new IsPhaseCondition(TurnPhase.COMBAT, false), // Only during combat
+                new InvertCondition(new DefendingPlayerControlsNoSourceCondition(filter)) // Only if defending player controls no snow land
+        );
+
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranGuardEffect(),
+                new TapSourceCost(),
+                condition,
+                "{T}: Target creature gets +1/+1 until end of turn. "
+                + "When that creature leaves the battlefield this turn, sacrifice {this}. "
+                + "Activate only during combat and only if defending player controls no snow lands."
+        );
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranGuard(final KjeldoranGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranGuard(this);
+    }
+}
+
+class KjeldoranGuardEffect extends OneShotEffect {
+
+    KjeldoranGuardEffect() {
+        super(Outcome.BoostCreature);
+        staticText = "Target creature gets +1/+1 until end of turn."
+                + "When that creature leaves the battlefield this turn, sacrifice {this}."
+                + "Activate only during combat and only if defending player controls no snow lands.";
+    }
+
+    private KjeldoranGuardEffect(KjeldoranGuardEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KjeldoranGuardEffect copy() {
+        return new KjeldoranGuardEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID targetId = source.getFirstTarget();
+        Permanent targetPermanent = game.getPermanent(targetId);
```

Comment:
No need for a targetId variable, inline it here and null check targetPermanent directly. If needed later cleaner to call by permanent.getId()